wire-library/wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt

```diff
@@ -768,15 +790,29 @@ class SwiftGenerator private constructor(
                     )
                   }
 
-                  if (field.typeName.optional) {
+                  val defaultValue = field.codableDefaultValue
+
+                  if (field.isOptional) {
+                    // A proto3 field that is defined with the optional keyword supports field presence.
+                    // Fields that have a value set and that support field presence always include the field value
+                    // in the JSON-encoded output, even if it is the default value.
                     beginControlFlow("if", "includeDefaults || self.%N != nil", field.name)
                     addEncode()
                     endControlFlow("if")
-                  } else if (field.isRepeated || field.isMap) {
+                  } else if (field.isCollection) {
                     beginControlFlow("if", "includeDefaults || !self.%N.isEmpty", field.name)
                     addEncode()
                     endControlFlow("if")
+                  } else if (field.isEnum) {
+                    beginControlFlow("if", "includeDefaults || self.%N.rawValue != 0", field.name)
+                    addEncode()
+                    endControlFlow("if")
+                  } else if (defaultValue != null) {
+                    beginControlFlow("if", "includeDefaults || self.%N != $defaultValue", field.name)
+                    addEncode()
+                    endControlFlow("if")
                   } else {
+                    // Message is fundamentally broken right now when it comes to evaluating "isDefault"
```

Comment:
This comment should probably explain how it's broken and what needs to be added.